File: Volumen2/problema217/P217_Viejo.java
package problema217;

import java.io.BufferedOutputStream;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStream;

public class P217_Viejo {

	public static void main(String[] args) throws IOException {

		BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
		OutputStream output = new BufferedOutputStream(System.out);

		int numero = Integer.parseInt(input.readLine());
		while (numero != 0) {
			boolean par = (numero & 1) == 0;
			output.write((par ? "DERECHA\n" : "IZQUIERDA\n").getBytes());
			numero = Integer.parseInt(input.readLine());
		}
		output.flush();
	}

}

File: Volumen2/problema217/P217_Viejo.cpp
#include <iostream>

int main()
{
    int numero;
    std::cin >> numero;
    while (numero != 0)
    {
        bool par = (numero & 1) == 0;
        std::cout << ((par) ? "DERECHA\n" : "IZQUIERDA\n");
        std::cin >> numero;
    }
    
    return 0;
}